fedimint-server/src/config.rs

```diff
@@ -155,27 +209,12 @@ impl ServerConfig {
             bail!("Peer ids are not indexed from 0");
         }
 
-        let res: anyhow::Result<Vec<()>> = self
-            .modules
-            .iter()
-            .map(|(name, cfg)| match name.as_str() {
-                "wallet" => cfg
-                    .to_typed::<WalletConfig>()
-                    .expect("valid config should not fail")
-                    .validate_config(identity),
-                "mint" => cfg
-                    .to_typed::<MintConfig>()
-                    .expect("valid config should not fail")
-                    .validate_config(identity),
-                "ln" => cfg
-                    .to_typed::<LightningModuleConfig>()
-                    .expect("valid config should not fail")
-                    .validate_config(identity),
-                other => panic!("module name that was not expected: {other}"),
-            })
-            .collect();
-
-        res?;
+        self.get_module_config::<WalletConfig>("wallet")?
+            .validate_config(identity)?;
+        self.get_module_config::<MintConfig>("mint")?
+            .validate_config(identity)?;
+        self.get_module_config::<LightningModuleConfig>("ln")?
+            .validate_config(identity)?;
```

Comment:
I'll make this generic over any set of modules soon. 

There is a bunch of work left to do to modularize the configgen (which I'm doing now). We should coordinate on when we work on which parts. As long as it's localized change it should work, just interface changes etc. would be annoying to rebase imo.